src/Domain/Donation.php

```diff
@@ -69,4 +109,98 @@ public function setBankData( BankData $bankData ) {
 		$this->bankData = $bankData;
 	}
 
+	public function getTracking(): string {
+		return $this->tracking;
+	}
+
+	public function setTracking( string $tracking ) {
+		$this->assertIsWritable();
+		$this->tracking = $tracking;
+	}
+
+	public function getSource(): string {
+		return $this->source;
+	}
+
+	public function setSource( string $source ) {
+		$this->source = $source;
+	}
+
+	public function getTotalImpressionCount(): int {
+		return $this->totalImpressionCount;
+	}
+
+	public function setTotalImpressionCount( int $totalImpressionCount ) {
+		$this->totalImpressionCount = $totalImpressionCount;
+	}
+
+	public function getSingleBannerImpressionCount(): int {
+		return $this->singleBannerImpressionCount;
+	}
+
+	public function setSingleBannerImpressionCount( int $singleBannerImpressionCount ) {
+		$this->singleBannerImpressionCount = $singleBannerImpressionCount;
+	}
+
+	public function getColor(): string {
+		return $this->color;
+	}
+
+	public function setColor( string $color ) {
+		$this->color = $color;
+	}
+
+	public function getSkin(): string {
+		return $this->skin;
+	}
+
+	public function setSkin( string $skin ) {
+		$this->skin = $skin;
+	}
+
+	public function getLayout(): string {
+		return $this->layout;
+	}
+
+	public function setLayout( string $layout ) {
+		$this->layout = $layout;
+	}
+
+	public function getInitialStatus(): string {
+		if ( $this->paymentType === PaymentType::DIRECT_DEBIT ) {
+			return self::STATUS_NEW;
+		}
+
+		if ( $this->paymentType === PaymentType::BANK_TRANSFER ) {
+			return self::STATUS_PROMISE;
+		}
+
+		return self::STATUS_EXTERNAL_INCOMPLETE;
+	}
+
+	public function determineFullName() {
+		if ( $this->getPersonalInfo() !== null ) {
+			return $this->getPersonalInfo()->getPersonName()->getFullName();
+		}
+
+		return 'Anonym';
+	}
+
+	public function generateTransferCode() {
```

Comment:
Where do you intend to use this? This code appears to have no cohesion with the Donation class, which suggests it does not belong here. Can't say without knowing what this is intended for though.